### DISCUSSION TITLE
Wrong cursor over controls on mac

### DISCUSSION THREAD

**Discussion by Pictony** [fabricjs/fabric.js#7643] [Q&A] [answered]
Hi!
I encounter a problem with cursors on Mac OS. Cursors on draggable rectangles are not the good ones, and cursors don't change when hovering controls. Theses probems occur with every browsers on Mac, not on PC. Any ideas?

**Comment by asturur** [comment]
No, i use a mac and i don't have this kind of issues.
Can you reproduce it on a jsfiddle?

**Reply by Pictony** [reply]
I didn't change any cursor properties, so the default fabricjs should apply.
Link of app is: [https://www.prisedetete.net/chapitre4-2.php](url)

I made some screenshots of the bugs here: [https://www.prisedetete.net/images/bugs-curseurs.jpg](url)
n°1: Expected behaviors ("move" and "nwse-resize" type) which are default fabricjs behaviors. Work as expecting on PC.
n°2: Behaviors on Chrome and Safari on Mac OS (online émulators).
n°3: Behaviors on Firefox on Mac OS (online emulator).
n°4: Resizing behavior on Chrome and Safari on Mac OS (test on real device).

It seems very random... Maybe I should try to overwrite default fabricjs cursors? (I don't know how to do that for controls...)

**Comment by asturur** [comment]
What is actually the bug? from the screenshot is not apparent.
Cursors are different, yes but that may depend on version of the broswers, especially for the resize one.
Not sure why you get a crosshair sometimes.

Did you manage to make it happen on a real mac outside emulators?

**Comment by chonigman-vt** [comment]
I'm having similar issues with cursor settings (CSS cursor in general) on a new M1 Mac. My co-worker's cursor kind of freaks out when the cursor attempts to change. Wondering @Pictony what machine and OS version you are using.

**Comment by Pictony** [comment, marked as answer]
I was able to test on real Mac yesterday evening. It's ok so I'm reassured (it's the same as my last screenshot which was on real mac too). It's kind of weird because cursor on Mac Os are not the same (resize cursor is not double arrow for example) and as a Pc user, I was not aware of this.
